.azure-pipelines/MacOS-CI.yml

```diff
@@ -40,76 +40,36 @@ jobs:
       versionSpec: '$(python.version)'
 
   - script: |
-      # Install Protobuf from source
       export NUM_CORES=`sysctl -n hw.logicalcpu`
       if [ '$(onnx_debug)' == '1' ]; then
-        export DEBUG=1
         source workflow_scripts/protobuf/build_protobuf_unix.sh $NUM_CORES $(pwd)/protobuf/protobuf_install Debug
       else
         source workflow_scripts/protobuf/build_protobuf_unix.sh $NUM_CORES $(pwd)/protobuf/protobuf_install
       fi
-
+    displayName: 'Install Protobuf from source'
+  - script: |
       git submodule update --init --recursive
-      python -m pip install -q --upgrade pip
-      python -m pip install -q -r requirements-release.txt
-
-      if [ '$(onnx_ml)' == '1' ]; then
-        export ONNX_ML=1
-      fi
-      export ONNX_BUILD_TESTS=1
-      export CMAKE_ARGS="-DONNX_WERROR=ON"
-      if [ '$(onnx_lite)' == '1' ]; then
-        export CMAKE_ARGS="${CMAKE_ARGS} -DONNX_USE_LITE_PROTO=ON"
-      fi
-      export ONNX_NAMESPACE=ONNX_NAMESPACE_FOO_BAR_FOR_CI
-      python setup.py -q install
-    displayName: 'Install dependencies and ONNX'
+    displayName: 'Load all submodules'
 
   - script: |
-      # check auto-gen files up-to-date
-      python onnx/defs/gen_doc.py
-      python onnx/gen_proto.py -l
-      python onnx/gen_proto.py -l --ml
-      python onnx/backend/test/stat_coverage.py
-
-      git status
-      git diff --exit-code  -- . ':(exclude)onnx/onnx-data.proto' ':(exclude)onnx/onnx-data.proto3'
-      if [ $? -ne 0 ]; then
-        echo "git diff returned failures"
-        exit 1
-      fi
-
-      # Do not hardcode onnx's namespace in the c++ source code, so that
-      # other libraries who statically link with onnx can hide onnx symbols
-      # in a private namespace.
-      ! grep -R --include='*.cc' --include='*.h' 'namespace onnx' .
-      ! grep -R --include='*.cc' --include='*.h' 'onnx::' .
+      pip install nox
```

Comment:
> @jcwchen Reverted image versions. Now only macos is failing because of the not found protobuf compiler. Any idea how I can fix that?

Cross scripts is sometimes a headache in AZP. I guess the export path in the first script for installing Protobuf from source was not applied in the following scripts? If yes, manually export in the this script before installing onnx like https://github.com/onnx/onnx/blob/e3bc0589b0f6277b8b347c0883d1c0402a60ea98/workflow_scripts/protobuf/build_protobuf_unix.sh#L35.

Reply:
I didn't realize there was a path exported. thanks!